src/main/scala-2.13+/org/apache/logging/log4j/scala/LoggingContext.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache license, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the license for the specific language governing permissions and
+ * limitations under the license.
+ */
+package org.apache.logging.log4j.scala
+
+import org.apache.logging.log4j.ThreadContext
+
+import scala.jdk.CollectionConverters._
+import scala.collection.{immutable, mutable}
+
+/** Manages the context data (context map, MDC) that is added to log events.
+  *
+  * A wrapper around `org.apache.logging.log4j.ThreadContext`.
+  */
+object LoggingContext extends mutable.Map[String, String] {
```

Comment:
Is this class added due to the collections API revamp in 2.13? I haven't been keeping up with that API at all.

Reply:
Yep, unluckily the mutable.Map in scala 2.13 has some methods that where `final`, and because they moved again the java-compat implicits I choose to keep things clean - notice also the new `IterableOnce` interface.
There is a scala collection compatibility but is only for 2.11 and 2.12, or just drop 2.10 and we can clean up all the multiple src folders.

Reply:
This looks very good, I did similar changes locally and have just a couple of comments: